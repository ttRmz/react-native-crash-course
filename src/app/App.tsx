import React from 'react';
import {useUserContext} from '../contexts/user';
import {AuthenticatedUser} from '../routes/AuthenticatedUser';
import {UnauthenticatedUser} from '../routes/UnauthenticatedUser';
import AppProviders from './AppProviders';

function App() {
  const {user} = useUserContext();

  return user ? <AuthenticatedUser /> : <UnauthenticatedUser />;
}

export default function AppWithProviders() {
  return (
    <AppProviders>
      <App />
    </AppProviders>
  );
}
