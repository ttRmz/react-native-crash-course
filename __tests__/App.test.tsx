import {render} from '@testing-library/react-native';
import React from 'react';
import App from '../src/app/App';

test('should render App correctly', () => {
  const {toJSON} = render(<App />);
  expect(toJSON()).toMatchSnapshot();
});

// test('should go to Explore page, then go back', () => {
//   const {queryByText, getByText} = render(<App />);
//   expect(queryByText('Go Back')).toBeFalsy();

//   fireEvent.press(getByText('Go to Explore page'));
//   expect(getByText('Go Back')).toBeTruthy();
//   expect(getByText('Explore')).toBeTruthy();

//   fireEvent.press(getByText('Go Back'));
//   expect(getByText('Home page ðŸ‘‹')).toBeTruthy();
//   expect(queryByText('Go Back')).toBeFalsy();
// });
