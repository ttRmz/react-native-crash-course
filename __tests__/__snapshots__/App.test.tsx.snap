// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "display": "flex",
                "height": 44,
                "justifyContent": "center",
                "position": "relative",
              }
            }
          >
            <Text>
              Login
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTScrollView
            contentContainerStyle={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
                "marginHorizontal": 44,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "fontWeight": "bold",
                    "justifyContent": "center",
                    "marginBottom": 16,
                  }
                }
              >
                Who are youÂ ?
              </Text>
              <TextInput
                collapsable={false}
                handlerTag={1}
                handlerType="NativeViewGestureHandler"
                onChangeText={[Function]}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "borderWidth": 1,
                    "height": 40,
                    "marginBottom": 32,
                    "padding": 10,
                    "width": "80%",
                  }
                }
                value="Useless Text"
              />
            </View>
          </RCTScrollView>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </RNCSafeAreaProvider>
</RCTSafeAreaView>
`;
